src/server/query.c

```diff
@@ -65,6 +66,58 @@ tds_lastpacket(TDSSOCKET * tds)
 	return tds->in_buf[1] != 0;
 }
 
+/**
+ * get query packet of a given type
+ * \tds
+ * \param head         extra information to put in a TDS7 header
+ */
+static TDSRET
+tds_get_query_head(TDSSOCKET * tds, TDSHEADERS * head)
+{
+	if (!IS_TDS72_PLUS(tds->conn)) {
+		return TDS_SUCCESS;
+	}
+
+	int qn_len = 0, more;
+	char *qn_msgtext = NULL;
+	char *qn_options = NULL;
+	size_t qn_msgtext_len = 0;
+	size_t qn_options_len = 0;
+
+	qn_len = tds_get_int(tds) - 4 - 18;             /* total length */
+	tds_get_int(tds);                          /* length: transaction descriptor, ignored */
+	tds_get_smallint(tds);                      /* type: transaction descriptor, ignored */
+	tds_get_n(tds, tds->conn->tds72_transaction, 8);  /* transaction */
+	tds_get_int(tds);                           /* request count, ignored */
+	if (qn_len != 0) {
+		if (!head) {
```

Comment:
would it not better to require `head` not to be NULL calling this function?

Reply:
removed test, head required